edbrowse fails to build

## Issue description

`edbrowse` fails to build with the following error:
```
gcc -DEDBROWSE_ON_LINUX -Wall   -c -o http.o http.c
http.c: In function ‘ebcurl_setError’:
http.c:1558:2: error: duplicate case value
  case CURLE_SSL_CACERT:
  ^~~~
http.c:1557:2: note: previously used here
  case CURLE_PEER_FAILED_VERIFICATION:
  ^~~~
make: *** [<builtin>: http.o] Error 1
```

The two constants have the same value and are the same branch in the switch.

### Steps to reproduce

`nix-build -p edbrowse` fails on my current channel, or you can use `nix-build -E 'let nixpkgs = import (builtins.fetchTarball { url = "https://github.com/NixOS/nixpkgs/archive/487cbfc5634e1f89eb4a72df9ce8e425a0c999f6.tar.gz"; sha256 = "1vw710y1g5b18xjiqhlc2pi3czr1b6zm8cqbigkqz9kxpqmb5awf"; }) {}; in nixpkgs.pkgs.edbrowse'`

## Technical details

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.

```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.18.0-13-generic, Ubuntu, 18.10 (Cosmic Cuttlefish)`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.1.3`
 - channels(endgame): `"nixpkgs-19.03pre164964.b0f40b78513"`
 - nixpkgs: `/home/endgame/.nix-defexpr/channels/nixpkgs`
```

--- a/pkgs/applications/editors/edbrowse/default.nix
+++ b/pkgs/applications/editors/edbrowse/default.nix
@@ -1,4 +1,4 @@
-{ stdenv, fetchFromGitHub, duktape, curl, pcre, readline, openssl, perl, html-tidy }:
+{ stdenv, fetchFromGitHub, fetchpatch, duktape, curl, pcre, readline, openssl, perl, html-tidy }:
 
 stdenv.mkDerivation rec {
   name = "edbrowse-${version}";
@@ -6,7 +6,15 @@ stdenv.mkDerivation rec {
 
   buildInputs = [ curl pcre readline openssl duktape perl html-tidy ];
 
-  patchPhase = ''
+  patches = [
+    # Fix build against recent libcurl
+    (fetchpatch {
+      url = https://github.com/CMB/edbrowse/commit/5d2b9e21fdf019f461ebe62738d615428d5db963.diff;
+      sha256 = "167q8n0syj3iv6lxrbpv4kvb63j4byj4qxrxayy08bah3pss3gky";
+    })
+  ];
+
+  postPatch = ''
     for i in ./tools/*.pl
     do
       substituteInPlace $i --replace "/usr/bin/perl" "${perl}/bin/perl"

